Makefile

```diff
@@ -0,0 +1,75 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect C++ sources, and link if necessary
+CPPSRC := $(wildcard src/*.cc)
+
+ifeq (, $(CPPSRC))
+	ADDITIONALLIBS := 
+else
+	ADDITIONALLIBS := -lc++
+endif
+
+# collect sources
+SRC := $(wildcard src/*.c)
+SRC += $(CPPSRC)
+OBJ := $(addsuffix .o,$(basename $(SRC)))
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+CPPFLAGS ?= -O3 -Wall -Wextra -Werror
+override CFLAGS += -std=gnu99
+override CPPFLAGS += -fPIC
```

Comment:
I'm not positive, but I think that to build a shared library, we may need to compile with `-fPIC` for both the C and C++ file, not just the C++ file.

I also think we may want to set the same default optimization level and warning settings for both C and C++ sources.

On a related note, @dcreager You had that comment about avoiding directly overriding the `CFLAGS`. I'm personally OK with making certain flags (like `-fPIC` and `-std=gnu99`) mandatory, if we don't think our sources will build correctly without them, but I may be missing something.

Reply:
If I understand this make syntax correctly, the only value that applies to only C sources is `-std=gnu99`. Everything else applies to both.

Reply:
Oh, of course! I was misreading this as `CXXFLAGS`, which is funny because I asked you to remove a comment explaining what this was.

This is great as-is.